crypto/fipsmodule/sha/keccak1600.c

```diff
@@ -356,16 +356,21 @@ size_t SHA3_Absorb(uint64_t A[SHA3_ROWS][SHA3_ROWS], const uint8_t *inp, size_t
     return len;
 }
 
- // SHA3_Squeeze is called once at the end to generate |out| hash value
- // of |len| bytes.
-void SHA3_Squeeze(uint64_t A[SHA3_ROWS][SHA3_ROWS], uint8_t *out, size_t len, size_t r)
+// SHA3_Squeeze can be called multiple times to incrementally 
+// generate |out| value of |len| bytes. |first| indicates weather 
+// it is the first call to SHA3_Squeeze.
+void SHA3_Squeeze(uint64_t A[SHA3_ROWS][SHA3_ROWS], uint8_t *out, size_t len, size_t r, int padded)
```

Comment:
Would you expect
```
SHA3_Squeeze(..., len = 4,...) || SHA3_Squeeze(..., len = 4, ...) = SHA3_Squeeze(..., len = 8, ...)
```
to be true?

Reply:
No, it will not be true. Callout: OpenSSL has support for this use-case (https://github.com/openssl/openssl/blob/af8b7e43e42c08ee5c94dacb1feec1284d47d06d/crypto/sha/sha3.c#L174).